### PR TITLE
WDFN-242 - Create a node.js graph server

### PR DESCRIPTION
**Pull request by danielnaab**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

WDFN-242
-----------
- Create a node.js server at `graph-server/index.js`
- Move Flask-specific things into `wdfn-server`
- Move asset-specific things into `assets`
- To stop abused node run scripts for cross-language tooling, create Makefiles for each project to create a dev environment. 

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: assets/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+else
+	@echo 'Please use node.js version $(NODE_VERSION).'
+	@echo 'Consider nvm to manage node versions: https://github.com/creationix/nvm'
+	@exit 1
+endif
+
+env-assets: assert-assets-node assets/node_modules
+	@echo 'Created assets environment!'
+
+assets/node_modules:
```
Comment (mbucknell) [inline assets/Makefile]: Caveat, I'm not a Makefile expert. Should installing the npm dependencies by part of env-assets?

Comment (danielnaab) [reply assets/Makefile]: Yep - here, `assets/node_modules` (the directory) is the build target. If the directory doesn't exist, its build target is called, which does the `npm install`.

File: graph-server/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// NOTE: We might want to maintain a pool of browser instances rather
+// than just a single one. Alternately, we could maintain a process pool of
+// node instances, and let the web server load balance between them.
+// For now, just use a single Chromium instance.
+puppeteer.launch().then((browser) => {
+    var server = http.createServer((req, res) => {
+        (async () => {
+            // Load a graph
+            const page = await browser.newPage();
+            await page.goto('http://localhost:5050/monitoring-location/07048495/');
```
Comment (mbucknell) [inline graph-server/index.js]: I have so wanted to try await/async !

**Comment by mbucknell** [comment]
Changes look good to me. You will ultimately need to update the repo README but this looks great!

File: README.md [hunk 3]
```diff
@@ -1,135 +1,42 @@
-# Water Data for the Nation UI
+# Water Data For The Nation UI
 
 [![Build Status](https://travis-ci.org/usgs/waterdataui.svg?branch=master)](https://travis-ci.org/usgs/waterdataui)
-[![Coverage Status](https://coveralls.io/repos/github/usgs/waterdataui/badge.svg?branch=master)](https://coveralls.io/github/usgs/waterdataui?branch=master)
+[![Coverage Status](https://coveralls.io/repos/github/u`wdfn-server`sgs/waterdataui/badge.svg?branch=master)](https://coveralls.io/github/usgs/waterdataui?branch=master)
```
Comment (mbucknell) [inline README.md]: I think this was changed by mistake.

**Comment by mbucknell** [comment]
I successfully built and ran everything. I ran each server in it's own window. It would be good if the README.md could describe how to run each one individually. Also could you explictly describe how to do a clean build?

File: README.md (hunk 3 above)
Comment (mbucknell) [reply README.md]: Is line 4 correct? It seems like a search/replace error.

Comment (danielnaab) [reply README.md]: Thanks, didn't mean to do that!

**Pull request merged**